From 8d916749b1f3b4479295749c0f299e28624efcd0 Mon Sep 17 00:00:00 2001
From: Hongzhang Yang <Hongzhang.Yang@freescale.com>
Date: Tue, 9 Jul 2013 19:39:38 +0800
Subject: [PATCH 402/505] ENGR00270296 bugfix: time stamp of 1st buffer to
 queue is 0

Bug:
Time stamp of 1st buffer to queue is 0

Root cause:
For the 1st buffer, disp->buf.timestamp is overwitten by
ioctl VIDIOC_QUERYBUF.

Solution:
Calculate disp->buf.timestamp after calling VIDIOC_QUERYBUF.

Signed-off-by: Hongzhang Yang <Hongzhang.Yang@freescale.com>
---
 test/mxc_vpu_test/display.c | 18 +++++++++---------
 1 file changed, 9 insertions(+), 9 deletions(-)

diff --git a/test/mxc_vpu_test/display.c b/test/mxc_vpu_test/display.c
index 9fd2020..64a04cd 100644
--- a/test/mxc_vpu_test/display.c
+++ b/test/mxc_vpu_test/display.c
@@ -939,15 +939,6 @@ int v4l_put_data(struct decode *dec, int index, int field, int fps)
 
 	disp = dec->disp;
 
-	if (disp->ncount == 0) {
-		gettimeofday(&tv, 0);
-		disp->buf.timestamp.tv_sec = tv.tv_sec;
-		disp->buf.timestamp.tv_usec = tv.tv_usec;
-
-		disp->sec = tv.tv_sec;
-		disp->usec = tv.tv_usec;
-	}
-
 	disp->buf.type = V4L2_BUF_TYPE_VIDEO_OUTPUT;
 	disp->buf.memory = V4L2_MEMORY_MMAP;
 
@@ -959,6 +950,15 @@ int v4l_put_data(struct decode *dec, int index, int field, int fps)
 		goto err;
 	}
 
+	if (disp->ncount == 0) {
+		gettimeofday(&tv, 0);
+		disp->buf.timestamp.tv_sec = tv.tv_sec;
+		disp->buf.timestamp.tv_usec = tv.tv_usec;
+
+		disp->sec = tv.tv_sec;
+		disp->usec = tv.tv_usec;
+	}
+
 	if (disp->ncount > 0) {
 		if (fps != 0) {
 			disp->usec += (1000000 / fps);
-- 
1.8.0

